src/neo/Cryptography/MerkleTree.cs

```diff
@@ -11,12 +11,12 @@ public class MerkleTree
     {
         private readonly MerkleTreeNode root;
 
-        public int Depth { get; private set; }
+        public int Depth { get; }
 
         internal MerkleTree(UInt256[] hashes)
         {
-            if (hashes is null || hashes.Length == 0) throw new ArgumentException();
             this.root = Build(hashes.Select(p => new MerkleTreeNode { Hash = p }).ToArray());
```

Comment:
if hashes it's null it will fault

Reply:
It's an internal method and all invocations pass non-null value.